### PR TITLE
feat(core): add sign out confirm modal

### LINKED ISSUE #4580: Signing out of an account lacks Confirmation dialog

### PR DESCRIPTION
**Pull request by JimmFly**
Close #4580 

https://github.com/toeverything/AFFiNE/assets/102217452/6f526678-bf2c-4d32-8253-8646b05e7a91

### CODE REVIEW

File: tests/affine-cloud/e2e/collaboration.spec.ts [hunk 1]
```diff
@@ -260,3 +262,23 @@ test.describe('collaboration members', () => {
...
+      {
+        name: 'test',
+      },
+      page
+    );
+    await clickSideBarCurrentWorkspaceBanner(page);
+    await page.getByTestId('workspace-modal-account-option').click();
+    await page.getByTestId('workspace-modal-sign-out-option').click();
+    await page.getByTestId('confirm-sign-out-button').click();
+    await clickSideBarCurrentWorkspaceBanner(page);
```
Comment (joooye34) [inline tests/affine-cloud/e2e/collaboration.spec.ts:280]: check the url redirect to index here

File: tests/kit/utils/cloud.ts [hunk 2]
```diff
@@ -176,8 +181,25 @@ export async function enableCloudWorkspace(page: Page) {
...
+  const shareMenuButton = page.getByTestId('local-share-menu-button');
+  expect(await shareMenuButton.isVisible()).toBeTruthy();
+  const maxAttempts = 5;
+  let attempt = 0;
+  let menuVisible = false;
+
+  while (!menuVisible && attempt < maxAttempts) {
+    try {
+      await shareMenuButton.click();
+      menuVisible = await page.getByTestId('local-share-menu').isVisible();
```
Comment (joooye34) [inline tests/kit/utils/cloud.ts:195]: why not use page.waitForSelector to check element exists ?

Comment (JimmFly) [reply tests/kit/utils/cloud.ts:195]: As I'm not quite sure whether the instability of this test is due to the need to wait for the appearance of the modal or the button being unresponsive, I am unable to reproduce this locally.

Comment (joooye34) [reply tests/kit/utils/cloud.ts:195]: try use waitForSelector ?

**Review by joooye34 — DISMISSED** [review]

**Review by joooye34 — APPROVED** [review]

**Pull request merged**